Build failure on NixOS: temporary value dropped while borrowed

```console
error: builder for '/nix/store/8i1j04z2k5g10miy0zc7371dabfrq4yd-hyprswitch.drv' failed with exit code 101;
       last 25 log lines:
       >    Compiling zbus_names v3.0.0
       >    Compiling zbus v4.4.0
       >    Compiling cairo-rs v0.20.1
       >    Compiling graphene-rs v0.20.1
       >    Compiling notify-rust v4.11.3
       >    Compiling pango v0.20.1
       >    Compiling gdk-pixbuf v0.20.1
       >    Compiling gdk4 v0.9.0
       >    Compiling gsk4 v0.9.0
       >    Compiling gtk4 v0.9.2
       >    Compiling gtk4-layer-shell v0.4.0
       >    Compiling hyprswitch v3.0.0 (/build/source)
       > error[E0716]: temporary value dropped while borrowed
       >   --> src/daemon/gui/gui.rs:85:100
       >    |
       > 85 |         let title = if show_title && !workspace.name.trim().is_empty() { &workspace.name } else { &workspace.id.to_string() };
       >    |             -----                                                                                  ^^^^^^^^^^^^^^^^^^^^^^^^ - temporary value is freed at the end of this statement
       >    |             |                                                                                      |
       >    |             borrow later stored here                                                               creates a temporary value which is freed while still in use
       >    |
       >    = note: consider using a `let` binding to create a longer lived value
       >
       > For more information about this error, try `rustc --explain E0716`.
       > error: could not compile `hyprswitch` (lib) due to 1 previous error
       > warning: build failed, waiting for other jobs to finish...
       For full logs, run 'nix log /nix/store/8i1j04z2k5g10miy0zc7371dabfrq4yd-hyprswitch.drv'.
error: 1 dependencies of derivation '/nix/store/1ph6r5652xc02a66805jffz375l42wnk-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/d59rv2j34ky9amrv0iyfhff4zhijn9bn-nixos-system-nixos-24.11.20241029.807e915.drv' failed to build
```

--- a/src/daemon/gui/gui.rs
+++ b/src/daemon/gui/gui.rs
@@ -82,7 +82,8 @@ pub(super) fn update(
         };
 
         let workspace_fixed = Fixed::builder().width_request(width).height_request(height).build();
-        let title = if show_title && !workspace.name.trim().is_empty() { &workspace.name } else { &workspace.id.to_string() };
+        let id_string = workspace.id.to_string();
+        let title = if show_title && !workspace.name.trim().is_empty() { &workspace.name } else { &id_string };
         let workspace_frame = Frame::builder().label(title).label_xalign(0.5).child(&workspace_fixed).build();
         let workspace_frame_overlay = Overlay::builder().css_classes(vec!["workspace", "background"]).child(&workspace_frame).build();
 

